crates/tree-morph/src/events.rs

```diff
@@ -0,0 +1,27 @@
+use uniplate::Uniplate;
+
+pub struct EventHandlers<T: Uniplate, M> {
+    pub on_enter: Vec<fn(&T, &mut M)>,
+    pub on_exit: Vec<fn(&T, &mut M)>,
+}
+
+impl<T: Uniplate, M> EventHandlers<T, M> {
+    pub fn new() -> Self {
+        EventHandlers {
+            on_enter: vec![],
+            on_exit: vec![],
+        }
+    }
+
+    pub fn trigger_on_enter(&self, node: &T, meta: &mut M) {
```

Comment:
Do we want these methods to be callable outside of tree-morph? If not, we should make these `pub(crate)` instead of `pub`